framework/mutex/FileMutex.php

```diff
@@ -115,15 +141,26 @@ protected function acquireLock($name, $timeout = 0)
      */
     protected function releaseLock($name)
     {
-        if (!isset($this->_files[$name]) || !flock($this->_files[$name], LOCK_UN)) {
+        if (!isset($this->_files[$name])) {
             return false;
-        } else {
+        }
+        
+        if (DIRECTORY_SEPARATOR === '\\') {
```

Comment:
This could be improved on readibility. The code itself should make most of the comments obsolete. I also prefer reuse of code to something like:
```php
$isWindows = DIRECTORY_SEPARATOR === '\\';
if($isWindows ){
  @unlink($this->getLockFilePath($name));
}
flock($this->_files[$name], LOCK_UN);
fclose($this->_files[$name]);
if(!$isWindows ){
  unlink($this->getLockFilePath($name));
}
```